drivers/gpio/gpio_pcal64xxa.c

```diff
@@ -891,6 +1047,16 @@ int pcal64xxa_init(const struct device *dev)
 		}
 	}
 
+#ifdef CONFIG_RETENTION_DRIVER_STATE
+	if (drv_cfg->retention_driver_state != NULL) {
+		rc = retention_driver_state_write(drv_cfg->retention_driver_state, NULL, 0);
```

Comment:
so the mechanism here is to use `retention_driver_state_write` with no data just for the `is_valid` bit of information, and then that's used to initialize back the state from the gpio registers itself, right?